src/libraries/System.Private.CoreLib/src/System/IO/FileStreamHelpers.Windows.cs

```diff
@@ -0,0 +1,158 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO
+{
+    // this type defines a set of stateless FileStreamStrategy helper methods
+    internal static class FileStreamHelpers
+    {
+        internal static FileStreamStrategy ChooseStrategy(FileStream fileStream, SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync)
+        {
+            // the switch exitst to measure the overhead of introducing a factory method to the FileStream ctor
+            // we are going to have more implementations soon and then it's going to make more sense
+            switch (isAsync)
+            {
+                case true:
+                    return new LegacyFileStreamStrategy(fileStream, handle, access, bufferSize, true);
+                case false:
+                    return new LegacyFileStreamStrategy(fileStream, handle, access, bufferSize, false);
+            }
```

Comment:
This switch just looks silly.  I know you explain it, but a switch here or not isn't going to impact anything measurably.  Just as in the Linux implementation, until there's a PR that adds additional strategies, this can just be:
```C#
=> new LegacyFileStreamStrategy(fileStream, handle, access, bufferSize, isAsync);
```